Tasks/PunishmentTasks.cs

```diff
@@ -57,6 +57,38 @@ public static async Task<bool> CheckMutesAsync()
                 }
 #if DEBUG
                 Program.discord.Logger.LogDebug(Program.CliptokEventID, "Checked mutes at {time} with result: {result}", DateTime.Now, success);
+#endif
+                return success;
+            }
+        }
+
+        public static async Task<bool> CheckAutomaticWarningsAsync()
+        {
+            if (Program.cfgjson.AutoWarnMsgAutoDeleteDays == 0)
+                return false;
+            Dictionary<string, UserWarning> warnList = Program.db.HashGetAll("automaticWarnings").ToDictionary(
+                x => x.Name.ToString(),
+                x => JsonConvert.DeserializeObject<UserWarning>(x.Value)
+            );
+            if (warnList is null | warnList.Keys.Count == 0)
+                return false;
+            else
+            {
+                // The success value will be changed later if any of the message deletes are successful.
+                bool success = false;
+                foreach (KeyValuePair<string, UserWarning> entry in warnList)
+                {
+                    UserWarning warn = entry.Value;
+                    if (DateTime.Now > warn.WarnTimestamp.AddDays(Program.cfgjson.AutoWarnMsgAutoDeleteDays))
+                    {
+                        var contextMessage = await DiscordHelpers.GetMessageFromReferenceAsync(warn.ContextMessageReference);
+                        await contextMessage.DeleteAsync();
+                        Program.db.HashDelete("automaticWarnings", warn.WarningId);
+                        success = true;
+                    }
+                }
+#if DEBUG
+                Program.discord.Logger.LogDebug(Program.CliptokEventID, "Checked automatic warnings at {time} with result: {result}", DateTime.Now, success);
 #endif
```

Comment:
Not needed, if you `LogDebug` in prod it will realise by itself based on the log level set at the beginning. I realise this is on the other tasks and will remove them after this
```suggestion
                Program.discord.Logger.LogDebug(Program.CliptokEventID, "Checked automatic warnings at {time} with result: {result}", DateTime.Now, success);
```